core/src/main/java/org/mule/runtime/core/internal/construct/AbstractPipeline.java

```diff
@@ -328,6 +332,11 @@ boolean isCompleteSignalRejectedExecutionException(Throwable throwable) {
       if (getStatistics().isEnabled()) {
         getStatistics().incReceivedEvents();
       }
+
+      FlowCallStack flowCallStack = event.getFlowCallStack();
+      if (flowCallStack instanceof DefaultFlowCallStack) {
```

Comment:
Shouldn't the push and pop methods be part of the interface? This will always be a stack.

Reply:
Only the runtime internals must push/pop, the exposed interface ir for read-only